devtools/lightning-inspect.js

```diff
@@ -843,18 +843,20 @@ window.attachInspector = function({Application, Element, ElementCore, Stage, Com
         updateTextureAttribs(this)
     }
 
-    const _updateFocus = Application.prototype.__updateFocus
-    Application.prototype.__updateFocus = function() {
-        const prev = this._focusPath && this._focusPath.length ? this._focusPath[this._focusPath.length - 1] : null;
-        _updateFocus.apply(this, arguments)
-        const focused = this._focusPath && this._focusPath.length ? this._focusPath[this._focusPath.length - 1] : null;
-
-        if (prev != focused) {
-            if (prev) {
-                val(prev, 'focused', false, false);
-            }
-            if (focused) {
-                val(focused, 'focused', true, false);
+    if (typeof Application !== "undefined") {
```

Comment:
What is the purpose of this change? I'd think maybe to avoid errors of Application is not defined, but the same errors can occur if the other classes are not defined.

Reply:
Never mind I see! Could you add a comment explaining why Application is optional?

Reply:
Well, `Component` was already optional, and we use neither `Component` nor `Application` - the other objects are core to the rendering so there's no reason to make them optional.